src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -798,7 +798,7 @@ namespace
             const int DefaultMaxDiffuseBounces = 3;
 
             const int max_bounces =
-                get_config<int>(config, construct_bounce_param_path(bounce_type), -1);
+                get_config<int>(config, construct_bounce_param_path(bounce_type), bounce_type == "diffuse" ? DefaultMaxDiffuseBounces : DefaultMaxBounces);
```

Comment:
While I _think_ this is correct, this function is getting really complicated and hard to follow.

What about adding a default value parameter to `load_separate_bounce_settings()`? That way checking the bounce type and deciding what default to use based on it would no longer be necessary.